packages/ai-core/src/browser/window-blink-service.ts

```diff
@@ -0,0 +1,257 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { injectable } from '@theia/core/shared/inversify';
+import { environment } from '@theia/core';
+
+/**
+ * Result of a window blink attempt
+ */
+export interface WindowBlinkResult {
+    /** Whether the window blink was successful */
+    success: boolean;
+    /** Error message if the blink failed */
+    error?: string;
+}
+
+/**
+ * Service for blinking/flashing the application window to get user attention.
+ */
+@injectable()
+export class WindowBlinkService {
```

Comment:
This does not work for the `window.titleBarStyle`: `custom` setting

Reply:
I changed the implementation of this and in my window-manager (i3wm) it works. Could be a follow-up if the problem remains.

Reply:
It still does not work for me. Follow up would be fine for me :+1: